tests/loom.rs

```diff
@@ -0,0 +1,48 @@
+#[cfg(loom)]
+#[cfg(test)]
+mod loom_tests {
+    use left_right::{Absorb, ReadGuard};
+    use loom::sync::atomic::AtomicUsize;
+    use loom::sync::atomic::Ordering::{Acquire, Relaxed, Release};
+    use loom::sync::Arc;
+    use loom::thread;
+
+    // TODO Would love to not re-define this!
```

Comment:
I think you could do that by making a subdirectory `common`, stick the struct + impls in `common/mod.rs`, and then place `mod common` at the top of this file.

Reply:
Hmmm, sorry, I'm confused there 😅. Since this test is in `tests/loom.rs` and not under `src`, where would `common` go? Under `src` or `tests` or neither and be at the same level? If I'm going to have `mod common;` in this file then I imagine it has to be under `tests/` as well but then can I access it in `src/` tests? Or should I move the `loom` tests to be under `src` somewhere?

Reply:
Ah, yes, if you also want to use it in unit tests in `src/`, you have to play a bit more of a trick. Specifically, you'd have to use the [`#[path]` attribute](https://doc.rust-lang.org/nightly/reference/items/modules.html#the-path-attribute) on the `mod` statement in `src/` to bring in the type from `tests/` (or vice-versa) or the [`include!` macro](https://doc.rust-lang.org/std/macro.include.html) to just directly dump in the source. It gets further complicated by the fact that the path to the trait is different in the two places. I think the way to go is probably to use `include!` and then place the `use` for the trait just above the call to `include!`.

Reply:
Okay, great, I've attempted this with 606d6c3046379a9d0b6b311871682499b307113f. Lemme know if that's what you had in mind! I tried to follow https://stackoverflow.com/a/44545091/2891835 but I think there was a circular dependency since `left_right` had the utilities as a test dependency but the utilities had `left_right` as a dependency so it could implement `Absorb` 🤷

Reply:
Ah, almost. I was proposing you use `include!` in _both_ `src/` and `test/`

Reply:
Ah! So like 98b54b76a0694e2e910bd33857e007148a207095?